alf/utils/checkpoint_utils.py

```diff
@@ -348,5 +352,32 @@ def save(self, global_step):
         torch.save(optimizer_state, f_path + '-optimizer')
         torch.save(replay_buffer_state, f_path + '-replay_buffer')
 
+        if self._global_step == -1:
```

Comment:
When will this `global_step` be equal to -1? In your test case, it starts with 0.

Reply:
> When will this `global_step` be equal to -1? In your test case, it starts with 0.
`global_step` is initialized as -1. Therefore, for the first time of checkpoint saving (``if self._global_step == -1``), we will also save the checkpoint structure. This is used to save the structure only once.